catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/SyncCatalogCommand.java

```diff
@@ -0,0 +1,135 @@
+import com.google.common.collect.Iterables;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.impl.SortByImpl;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.util.impl.ResultIterable;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.api.console.Session;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.codice.ddf.security.Security;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.client.solrj.UnavailableSolrException;
+import org.codice.solr.factory.SolrClientFactory;
+import org.opengis.filter.sort.SortOrder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+@Command(
+  scope = "offline-solr-gazetteer",
+  name = "synccatalog",
+  description = "Syncs all catalog items to the offline solr gazetteer core"
+)
+public class SyncCatalogCommand implements Action {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(RemoveAllOfflineSolrGazetteerCommand.class);
+  public static final int PARTITION_SIZE = 256;
+
+  @Reference private SolrClientFactory clientFactory;
+
+  @Reference private Session session;
+
+  @Reference private CatalogFramework catalogFramework;
+  @Reference private FilterBuilder filterBuilder;
+  @Reference private Security security;
+
+  private final RetryPolicy retryPolicy =
+      new RetryPolicy()
+          .retryOn(ImmutableList.of(UnavailableSolrException.class, SolrServerException.class))
+          .withMaxDuration(5, TimeUnit.SECONDS)
+          .withBackoff(50, 1_000, TimeUnit.MILLISECONDS);
+
+  @Override
+  public Object execute() throws Exception {
+    return security.runWithSubjectOrElevate(this::executeWithSubject);
+  }
+
+  public Object executeWithSubject() throws Exception {
+    SolrClient solrClient =
+        clientFactory.newClient(OfflineGazetteerPlugin.STANDALONE_GAZETTEER_CORE_NAME);
+
+    Failsafe.with(retryPolicy).get(() -> solrClient.ping());
+
+    Iterable<Result> iterable =
+        ResultIterable.resultIterable(catalogFramework, getGazetteerFilter());
+
+    session.getConsole().println("Starting sync...");
+    long counter = 0;
+    Instant start = Instant.now();
+
+    for (List<Result> results : Iterables.partition(iterable, PARTITION_SIZE)) {
+      if (Thread.interrupted()) {
```

Comment:
👍  Yes. Absolutely.

Reply:
I was not aware of this.

https://github.com/apache/karaf/blob/cffa8250da16c9be9023ea5abeaddfc3bae4e746/shell/console/src/main/java/org/apache/karaf/shell/console/AbstractAction.java#L39-L49

Reply:
@pklinef Would you prefer using that built in method?

Reply:
I do not have a strong preference but might as well use it just in case they add future fixes to it.  For example, what does the additional `yield` provide?

Reply:
I think i'm inclined to say the opposite. Particularly because of the `yield`. Most literature aligns with the javadoc that yield should rarely ever be used except in specific scenarios (coupled with thorough profiling and testing). Using it in a common utility method seems to go against those conditions. Let alone with i'm not really sure what `yield` would accomplish in this scenario. If something were polling (or some kind of busy wait loop) it might make more sense.

Reply:
Hard to say since this is probably a thread running under JLine which handles console key events like crtl+c. 

I did see other examples in Karaf where they do not do the `Thread.yield()` and only do `Thread.interrupted()`.

I checked the commit but did not find any more details on why that method includes `Thread.yield()`.
https://github.com/apache/karaf/commit/220e260c51c723f368fa81557f41cf20a009b2a0

Here is the other example where they do not use `Thread.yield()`.
https://github.com/apache/karaf/blob/cffa8250da16c9be9023ea5abeaddfc3bae4e746/shell/core/src/main/java/org/apache/karaf/shell/impl/console/commands/Procedural.java#L566-L569

Looks like JLine also does it both ways.  Same developers contributing to JLine and Karaf though so not surprising.
https://github.com/search?q=org%3Ajline+Thread+isInterrupted&type=Code

I also noticed they use `Thread.currentThread().isInterrupted()` but `Thread.isInterrupted()` does the same thing since Java 6.
https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/jdk/src/share/classes/java/lang/Thread.java#L946-L948

Found this article that gives some of the OS level details about what `Thread.yield()` does but it also concludes you should practically never use it.
https://www.javamex.com/tutorials/threads/yield.shtml